### ISSUE TITLE
Cannot read property 'dep' of undefined

### ISSUE THREAD

**Issue by MartinMalinda** [vuejs/composition-api#480]
```
TypeError: Cannot read property 'dep' of undefined
  File "app:///vue-composition-api.common.prod.js", line 15, in be
    '{snip} e._watchers.length-1];return Q(j.value)&&g&&j.value.__ob__.dep.addSub({update:function(){j.run()}}),ye(j,c),function(){w()}}var ge={};var he {snip}
  File "app:///vue-composition-api.common.prod.js", line 15, in Object.exports.watch
    '{snip} return w({immediate:!1,deep:!1,flush:"post"},e)}(n);return be(pe(),e,r,o)},exports.watchEffect=function(e,t){var n=function(e){return w({imm {snip}
  File "app:///pinia.common.js", line 171, in buildStore
    compositionApi.watch(function () { return state.value; }, function (state) {
  File "app:///pinia.common.js", line 261, in useStore
    stores.set(id, (store = buildStore(id, state, getters, actions, getInitialState(id))));
  File "server.js", line 1, in setup
...
(5 additional frame(s) were not displayed)
```

This happens with Nuxt SSR, in production build only, so I'm kinda lost what is the source of the issue - if it's vue-composition-api or Nuxt or Pinia and how to debug it.

But it did start to happen with update of @vue/composition api from 0.5.X to >=1.0.0.

**Comment by danielroe** [comment]
Duplicate of https://github.com/vuejs/composition-api/issues/473, I think.

**Comment by MartinMalinda** [comment]
Ah. But yeah I updated to `"@vue/composition-api": "1.0.0-beta.9",` and still experiencing this.

**Comment by pikax** [comment]
This should be fixed on [1.0.0-beta.10](https://github.com/vuejs/composition-api/blob/master/CHANGELOG.md#100-beta10-2020-08-15) 

If not fixed, please reopen the issue and add a minimum reproduction repo or codesandbox
